packages/starlight/__tests__/basics/slugs.test.ts

```diff
@@ -76,3 +77,32 @@ describe('localizedSlug', () => {
 		expect(localizedSlug('test', undefined)).toBe('test');
 	});
 });
+
+describe('urlToSlug', () => {
+	test('returns slugs with `build.output: "directory"`', () => {
+		expect(urlToSlug(new URL('https://example.com/slug'))).toBe('slug');
+		expect(urlToSlug(new URL('https://example.com/dir/page/'))).toBe('dir/page');
+		expect(urlToSlug(new URL('https://example.com/dir/sub-dir/page/'))).toBe('dir/sub-dir/page');
+	});
+
+	test('returns slugs with `build.output: "file"`', () => {
+		expect(urlToSlug(new URL('https://example.com/slug.html'))).toBe('slug');
+		expect(urlToSlug(new URL('https://example.com/dir/page/index.html'))).toBe('dir/page');
+		expect(urlToSlug(new URL('https://example.com/dir/sub-dir/page.html'))).toBe(
+			'dir/sub-dir/page'
+		);
+	});
```

Comment:
Might also be worth testing `https://example.com/` and `https://example.com/index.html`?